src/components/plots/xyPlot.ts

```diff
@@ -71,6 +71,49 @@ module Plottable {
       return this;
     }
 
+    protected _updateExtentsForAttr(attr: string) {
+      if (attr === "x") {
+        return super._updateExtentsForAttr(attr, this._xFilter());
+      } else if (attr === "y") {
+        return super._updateExtentsForAttr(attr, this._yFilter());
+      }
+      return super._updateExtentsForAttr(attr);
+    }
```

Comment:
The tricky part is really you need to apply a filter inside `_computeExtent`, but you don't know what the filter is unless you have a the key being set. Maybe we should have a `D3.Map` of filters, and just try to apply the filter for a particular attribute every time? This way we don't need to override `_updateExtentsForAttr()` on `XYPlot`.